cmd/contour/serve.go

```diff
@@ -352,6 +383,92 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	}
 	g.Add(debugsvc.Start)
 
+	// Leader Election starts here.
+
+	leaderOK := make(chan struct{})
+	deposed := make(chan struct{})
+	leadingDisabled := make(chan struct{})
+
+	g.Add(func(stop <-chan struct{}) error {
+		log := log.WithField("context", "leaderelect")
+		if ctx.DisableLeaderElection {
+			log.Info("Leader election disabled")
+			// if leader election is disabled, signal the gRPC goroutine
+			// to start serving and block this goroutine forever.
+			close(leaderOK)
+			<-leadingDisabled
+		}
+
+		log.Info("Started leader election")
+		log.Infof("Leader election params: %#v", ctx.LeaderElectionConfig)
+
+		// Generate the event recorder to send election events to the logs.
+		// Set up the event bits
+		eventBroadcaster := record.NewBroadcaster()
+		eventBroadcaster.StartLogging(log.Infof)
```

Comment:
how much does this log? People complain when contour outputs periodic information.

Reply:
When the state changes only, no periodic logs.